Kernel/Random.h

```diff
@@ -120,6 +122,36 @@ class KernelRng : public FortunaPRNG<Crypto::Cipher::AESCipher, Crypto::Hash::SH
     KernelRng();
 };
 
+class EntropySource {
+    template<typename T>
+    struct Event {
+        u64 timestamp;
+        size_t source;
+        T event_data;
+    };
+
+public:
+    EntropySource()
+        : m_source(next_source++)
+    {
+    }
+
+    template<typename T>
+    void add_random_event(const T& event_data)
+    {
+        // we don't lock this because on the off chance a pool is corrupted, entropy isn't lost
```

Comment:
The general format of the comments here is similar to a sentence, e.g. `// We don't [...] entropy isn't lost.`